src/parser.ts

```diff
@@ -292,46 +295,50 @@ export class Parser {
       contents = this.parseFragment(opts, format);
     }
 
-    const endTok = this._t.peek();
-    const pos = this.getPos(startTok);
-    const end = this.getPos(endTok);
+    const nextTok = this._t.peek();
+    const start = this.getPos(startTok);
 
     // fragment ended but we don't have a close format. Convert this node into a text node.
-    if (endTok.name !== format) {
-      unshiftOrJoin(contents, this.finish({ name: 'text', contents: startTok.contents }, pos, end));
+    if (nextTok.name !== format) {
+      const lastTok = contents[contents.length - 1] ?? startTok;
+      unshiftOrJoin(
+        contents,
+        this.finish({ name: 'text', contents: startTok.contents }, start, lastTok.location.end)
+      );
       return contents;
-    } else {
-      this._t.next(); // consume end format.
-
-      if (contents.length === 0) {
-        // empty formats not allowed
-        return [
-          this.finish({ name: 'text', contents: startTok.contents + endTok.contents }, pos, end),
-        ];
-      } else if (format === 'tick') {
-        contents = contents.map(child =>
-          child.name === 'tag'
-            ? { ...child, name: 'text', contents: escapeHtml(child.contents) }
-            : child
-        );
-      } else if (format === 'pipe') {
-        const ntNode = parseNonTerminal(contents[0].contents);
-
-        if (ntNode === null) {
-          // failed to parse a non-terminal, so convert to text.
-          const firstPos = this.getPos(contents[0]);
-          const lastEnd = this.getEnd(contents[contents.length - 1]);
-          unshiftOrJoin(contents, this.finish({ name: 'text', contents: '|' }, pos, firstPos));
-          pushOrJoin(contents, this.finish({ name: 'text', contents: '|' }, lastEnd, end));
+    }
```

Comment:
Note that this is just a whitespace change - I converted the if/else to an early return. I recommend viewing with whitespace changes surpressed.